core/src/main/java/org/sonarsource/sonarlint/core/log/MessageFormat.java

```diff
@@ -0,0 +1,491 @@
+      str.append(Arrays.toString((boolean[]) o));
+    } else if (oClass == char[].class) {
+      str.append(Arrays.toString((char[]) o));
+    } else {
+      if (dejaVu == null) {
+        dejaVu = new HashSet<>();
+      }
+      // special handling of container Object[]
+      final String id = identityToString(o);
+      if (dejaVu.contains(id)) {
+        str.append(RECURSION_PREFIX).append(id).append(RECURSION_SUFFIX);
+      } else {
+        dejaVu.add(id);
+        final Object[] oArray = (Object[]) o;
+        str.append('[');
+        boolean first = true;
+        for (final Object current : oArray) {
+          if (first) {
+            first = false;
+          } else {
+            str.append(", ");
+          }
+          recursiveDeepToString(current, str, new HashSet<>(dejaVu));
+        }
+        str.append(']');
+      }
+      // str.append(Arrays.deepToString((Object[]) o));
+    }
+  }
+
+  private static void appendMap(final Object o, final StringBuilder str, @Nullable Set<String> dejaVu) {
+    // special handling of container Map
+    if (dejaVu == null) {
+      dejaVu = new HashSet<>();
+    }
+    final String id = identityToString(o);
+    if (dejaVu.contains(id)) {
+      str.append(RECURSION_PREFIX).append(id).append(RECURSION_SUFFIX);
+    } else {
+      dejaVu.add(id);
+      final Map<?, ?> oMap = (Map<?, ?>) o;
+      str.append('{');
+      boolean isFirst = true;
+      for (final Object o1 : oMap.entrySet()) {
+        final Map.Entry<?, ?> current = (Map.Entry<?, ?>) o1;
+        if (isFirst) {
+          isFirst = false;
+        } else {
+          str.append(", ");
+        }
+        final Object key = current.getKey();
+        final Object value = current.getValue();
+        recursiveDeepToString(key, str, new HashSet<>(dejaVu));
+        str.append('=');
+        recursiveDeepToString(value, str, new HashSet<>(dejaVu));
+      }
+      str.append('}');
+    }
+  }
+
+  private static void appendCollection(final Object o, final StringBuilder str, @Nullable Set<String> dejaVu) {
+    // special handling of container Collection
+    if (dejaVu == null) {
+      dejaVu = new HashSet<>();
+    }
+    final String id = identityToString(o);
+    if (dejaVu.contains(id)) {
+      str.append(RECURSION_PREFIX).append(id).append(RECURSION_SUFFIX);
+    } else {
+      dejaVu.add(id);
+      final Collection<?> oCol = (Collection<?>) o;
+      str.append('[');
+      boolean isFirst = true;
+      for (final Object anOCol : oCol) {
+        if (isFirst) {
+          isFirst = false;
+        } else {
+          str.append(", ");
+        }
+        recursiveDeepToString(anOCol, str, new HashSet<>(dejaVu));
+      }
+      str.append(']');
+    }
+  }
+
+  private static void tryObjectToString(final Object o, final StringBuilder str) {
+    // it's just some other Object, we can only use toString().
+    try {
+      str.append(o.toString());
+    } catch (final Throwable t) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Catch Exception instead of Throwable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1181)